### PR TITLE
ddl: make CI TestDropSchemaWithPartitionTable stable

### LINKED ISSUE #28206: Unstable `db_partition_test.go: testIntegrationSuite5.TestDropSchemaWithPartitionTable`
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_check_2/detail/tidb_ghpr_check_2/34569/pipeline

### 2. What did you expect to see? (Required)

Passing tests

### 3. What did you see instead (Required)

```bash
[2021-09-18T12:46:41.405Z] FAIL: db_partition_test.go:2940: testIntegrationSuite5.TestDropSchemaWithPartitionTable
[2021-09-18T12:46:41.405Z] 
[2021-09-18T12:46:41.405Z] db_partition_test.go:2990:
[2021-09-18T12:46:41.405Z]     c.Assert(recordsNum, Equals, 0)
[2021-09-18T12:46:41.405Z] ... obtained int = 4
[2021-09-18T12:46:41.405Z] ... expected int = 0
```

### 4. What is your TiDB version? (Required)

#28204 of commit 044e4b7425937600db3d1b441082f37e748f075b

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #28206

Problem Summary:

Fix a unstable test

### What is changed and how it works?

Wait the gc_delete_range work finish before check the data removed

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by AilinKid — APPROVED** [review]

File: ddl/db_partition_test.go [hunk 1]
```diff
@@ -2994,6 +3001,17 @@ func (s *testIntegrationSuite5) TestDropSchemaWithPartitionTable(c *C) {
 	})
 	c.Assert(err, IsNil)
 
+	startTime := time.Now()
+	done := waitGCDeleteRangeDone(c, tk, tableIDs[2])
+	if !done {
+		// Takes too long, give up the check.
+		logutil.BgLogger().Info("drop schema",
+			zap.Int64("id", tableIDs[0]),
+			zap.Stringer("duration", time.Since(startTime)),
```
Comment (AilinKid) [inline ddl/db_partition_test.go:3010]: For async test, the wait time is underestimated/uncontrollable when the CI is with a high workload. Better way is to sync it or move it out of unit tests.

Comment (tiancaiamao) [reply ddl/db_partition_test.go:3010]: Agree, in my local env, it's fast and never run here.
When it's slow in CI and fails, the log is helpful for debugging.

Comment (AilinKid) [reply ddl/db_partition_test.go:3010]: make sense

Comment (xhebox) [reply ddl/db_partition_test.go:3010]: Can not we set the logging as `Debug` ?

Comment (tiancaiamao) [reply ddl/db_partition_test.go:3010]: Hey, buddy, don't be too serious... it's the test code and never affect the user. @xhebox 
The default loglevel in CI is not set to `Debug`, so if it really happen we will miss the information.

**Review by xhebox — APPROVED** [review]

**Pull request merged**